### PR TITLE
AK - add sourceRepo to systeminfo

### LINKED ISSUE #2: add sourceRepo to systeminfo
The file `SystemInfo.java` has a plain old Java Object that represents global information about the app.

Currently, the repo that appears in the footer is hardcoded. For example at the bottom of each page, we see:

Image

And the link `GitHub` (as in "check out the source on GitHub") currently goes to:

https://github.com/ucsb-cs156-s22/s22-6pm-happycows

Instead, we want to specify that repo in the application.properties file, and be able to override it with a config var called `SOURCE_REPO`.

- [ ]  Add `String sourceRepo` to the `SystemInfo` class
- [ ]  Add initialization of this value from a property called app.sourceRepo in `src/main/resources/application.properties`, with the current default being: https://github.com/ucsb-cs156/proj-courses which is the main repo where the code that carries over from quarter to quarter is stored.
- [ ]  Allow that property to be overridden

 by an environment variable SOURCE_REPO
You can use app.admin.emails=${ADMIN_EMAILS:${env.ADMIN_EMAILS:[phtcon@ucsb.edu](mailto:phtcon@ucsb.edu)}} in /src/main/resources/application.properties as a model for how to proceed.

Then make sure that the correct value is exposed by looking at the `/api/systemInfo` endpoint in Swagger.

In this particular issue, you do NOT need to implement the frontend changes; that's separate.

### PR DESCRIPTION
**Pull request by Gopu2001**
Closes #2 

# Implementation Notes:
 * modified `SystemInfo.java` to incorporate the new variable
 * add the variable to the system implementation
 * set a default value of https://github.com/ucsb-cs156/proj-courses in `application.properties`
 * modified test for SystemInfo Controller

### CODE REVIEW

**Review by josiahnross — APPROVED** [review]
LGTM!

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/SystemInfoControllerTests.java [hunk 1]
```diff
@@ -51,6 +51,7 @@ public void systemInfo__admin_logged_in() throws Exception {
         .builder()
         .showSwaggerUILink(true)
         .springH2ConsoleEnabled(true)
+        .sourceRepo("https://github.com/ucsb-cs156/proj-courses")
```
Comment (Gopu2001) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/SystemInfoControllerTests.java:54]: Modify the test in the service implementation testing as well

**Review by Gopu2001 — COMMENTED** [review]
I need to make changes.

**Review by pconrad — APPROVED** [review]
LGTM... I will try to test it on your team's QA site before merging.

Nice job!

**Pull request merged**